pkg/upath/safejoin.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2021 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build !windows
+// +build !windows
+
+package upath
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+)
+
+// SafeFilepathJoin safely joins two paths path1+path2. The resulting path will
+// always be contained within path1 even if path2 tries to escape with "../".
+// If that path is not possible, an error is returned. The resulting path is
+// cleaned.
+func SafeFilepathJoin(path1, path2 string) (string, error) {
+	relPath, err := filepath.Rel(".", path2)
```

Comment:
this will return an error, if `path2` is absolute,  as it cant make relative between current dir and a absolute dir. Do you foresee that being an issue ? --  that does not look like a zipslip, maybe I am wrong

https://go.dev/play/p/NyuIS9wZ7M4

Reply:
The issue here is not with `path2` being absolute but rather the basepath being `"."` which would require knowledge of the working directory: `An error is returned if targpath can't be made relative to basepath or if knowing the current working directory would be necessary to compute it.`

Reply:
```suggestion
	relPath, err := filepath.Rel(os.Getwd(), path2)
```

Maybe this could fix it but I'd have to test it first.

Reply:
if you look at the test, absolute paths fail, so I'm not sure I see the meaning. 
Please refer to my new pull request from now on.